cors/access-control-expose-headers-parsing.window.js

```diff
@@ -1,13 +1,15 @@
-function exposeTest(resource, desc) {
-  const url = new URL("resources/" + resource, location.href).href.replace("://", "://élève.");
+promise_test(() => fetch("resources/access-control-expose-headers.json").then(res => res.json()).then(runTests), "Loading JSON…");
 
-  promise_test(() => {
-    return fetch(url).then(res => {
-      assert_equals(res.headers.get("content-language"), "sure")
-      assert_equals(res.headers.get("x-custom"), null);
-    })
-  }, "Access-Control-Expose-Headers parsing: " + desc);
+function runTests(allTestData) {
+  allTestData.forEach(testData => {
+    const encodedInput = encodeURIComponent(testData.input);
+    promise_test(() => {
+      const relativeURL = "resources/expose-headers.py?expose=" + encodedInput,
+            url = new URL(relativeURL, location.href).href.replace("://", "://élève.");
+      return fetch(url).then(res => {
+        assert_equals(res.headers.get("content-language"), "mkay");
+        assert_equals(res.headers.get("bb-8"), (testData.exposed ? "hey" : null));
+      });
+    }, "Parsing: " + encodedInput);
```

Comment:
I much prefer reading the test output if you don't encode it here. Reads so much nicer, and the descriptions are mostly meant for human consumption, and it's easier to see what it actually tests. If you want to debug the thing, you can figure out the exact thing anyway.
```suggestion
    }, "Parsing: " + testData.input);
```